projects/noloan/firebase/src/main/java/hasadna/noloan/firebase/DbMessages.java

```diff
@@ -123,5 +112,26 @@ public void notifyListeners(int index, SmsMessage smsMessage, Type type) {
     // SmsMessage parameter is passed for cases the index already had been removed from the db list
     void messageRemoved(int index, SmsMessage smsMessage);
   }
+
+  // static function to search a given list
+  public static int findSmsByBody(SmsMessage sms, ArrayList<SmsMessage> list) {
+    for (int i = 0; i < list.size(); i++) {
+      if (list.get(i).getBody().contentEquals(sms.getBody())
+          && list.get(i).getSender().contentEquals(sms.getSender())) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  public static int findSmsById(SmsMessage sms, ArrayList<SmsMessage> list) {
+    for (int i = 0; i < list.size(); i++) {
+      if (list.get(i).getId().contentEquals(sms.getId())) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
```

Comment:
I think that if it is important for you to keep `DbMessages` a class that handles only DB messages - These functions should be somewhere else, as they were added (if I understand correctly) for the purpose of handling inbox messages, but it's done via the `DbMessages` instance.

So I think for consistency, either:
1. Move them somewhere else - Perhaps `Util.java` ?
2. Switch back to `s/DbMessages/SmsMessages` model, having it for the purpose of handling both Inbox & DB messages. IMO, this solves your question of where to put the `Util.java`'s code as well.
https://github.com/hasadna/monorepo/blob/4580d23fe46c62a38f4b62975c0e86068aa8f1f7/projects/noloan/app/src/main/java/hasadna/noloan/Util.java#L14

BTW - This is the reason that the idea of having `s/DbMessages/SmsMessages` handle both Inbox & DbMessages raised up.

Reply:
I don't know where is a better place to put it.
Util is only in the app.

Reply:
So perhaps `s/DbMessages/SmsMessages`
Have it as a class that holds all functions handling SmsMessages + Holds the DB messages.
And the inbox list stays the same "under" the `InboxRecyclerAdapter`.

>Util is only in the app.

Having this ( https://github.com/hasadna/monorepo/pull/246#discussion_r344373476  ) and needing SmsMessages general functions for the admin app as well (In the admin app you don't use it by the name "inbox" but you use it for other local lists - suggested / approved)

Not having `s/DbMessages/SmsMessages` means you'll need to duplicate some functions to the admin app - I don't think this serves the purpose of "shared modules" you proposed and this whole change.

Not having `s/DbMessages/SmsMessages` and not duplicating them, means you should have `Util` be another shared module as well - IMO this is going a little too far with this idea.

> We can create another class for the inbox but I don't think it is useful.

I agree.

Reply:
>  means you'll need to duplicate some functions to
That is the reason it is static under the DbMessages, So we get access to it from both admin and the app...

Reply:
I still see it as you can get the `DbMessages` instance from both the admin app and the NoLoan app.

But cool, now I understand how you separate things, in terms of functions that aren't used by the `DbMessages` itself, and for external classes usage - static.
:+1: